### PR TITLE
563-refactor: Widget school menu

### LINKED ISSUE #269: Rework hover state for Links
### Description

Research and improve the hover state for the items in dropdown of header

### Prerequisites

Insufficient contrast when hovering over a link

Image

### LINKED ISSUE #563: FSD: Widget School-menu
## 📝 Describe the refactoring task

Part of the issue #347 

## 📍 Code location

`src/widgets/school-menu`

### school-menu.module.scss
1. Refactor `school-menu.tsx`'s `scss` as following

-   remove all parent selector `& .card-content` -> `.card-content`. 
-   change all possible values to variables if such exists.
-   and everything in your opinion that needs to be refactored
-   replace all element selectors by `tag` with `class` selector `img` -> `.image`

### school-menu.tsx
1. replace `div`'s with `HTML5` tags
2. replace `interface` with `type`
3. move `getMenuItem` to `school-menu/helpers`
4. replace `template string` in `className` with [classnames](https://www.npmjs.com/package/classnames)
5. move `school-menu` component and all related pieces to `school-menu/ui/school-menu`

### school-item.tsx
1. replace `interface` with `type`
2. replace `ternary operator` in `className` with [classnames](https://www.npmjs.com/package/classnames)
3. place all the object property checks in constants
4. extract styles from `school-item`'s `scss` and create separate `scss` file for `school-item` component

### school-list.tsx 
1. replace all the `interfaces` with `types`
2. replace complex className constant with [classnames](https://www.npmjs.com/package/classnames)
3. extract styles from `school-item`'s `scss` and create separate `scss` file for `school-list` component

### school-list.test.tsx 
1. replace all the links with constants

### constants.ts
1. move constants to `constants.ts`
2. move all the anchor links to the `shared ANCHORS constant`

### PR DESCRIPTION
**Pull request by Quiddlee**
## What type of PR is this? (select all that apply)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [x] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Related Tickets & Documents

- Related Issue #563 
- Closes #563 
- Related Issue #269 
- Closes #269 

## Screenshots, Recordings
Screenshot 2024-11-22 at 11 33 41 PM

## Added/updated tests?

- [x] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

## [optional] What gif best describes this PR or how it makes you feel?
gif

## Release Notes

- **New Features**
  - Introduced `communityMenuStaticLinks` and `schoolMenuStaticLinks` for enhanced menu navigation.
  - Updated `DesktopView` and `MobileView` components to utilize new static links for menu items.
  - Added new `SchoolMenu` component to render school items with customizable headings and colors.
  - New `Color` type for enforcing specific color options.

- **Bug Fixes**
  - Improved responsiveness of the mobile menu by adjusting height properties.

- **Tests**
  - Added comprehensive tests for the `SchoolMenu` and desktop menu functionality.

- **Styles**
  - Introduced new styles for `school-item` and `school-menu` components to enhance layout and responsiveness.

### CODE REVIEW

**Comment by Quiddlee** [comment]
run visual now

**Comment by SpaNb4** [comment]
Would be nice to take screenshots for the desktop menu as well
https://percy.io/8d310036/web/rs-school-standalone/builds/37627699/changed/2047524732?browser=chrome&browser_ids=59%2C60%2C62%2C63&category=unchanged&device_name=&group_snapshots_by=similar_diff&os=&other-browser=&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=375&widths=375%2C1280

File: src/widgets/school-menu/ui/school-menu/school-menu.test.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    links.forEach((link) => {
+      expect(link).toBeInTheDocument();
+    });
+
+    const descriptions = container.getElementsByTagName('small');
+
+    for (const description of descriptions) {
+      expect(description).toBeInTheDocument();
+    }
```
Comment (SpaNb4) [inline src/widgets/school-menu/ui/school-menu/school-menu.test.tsx:59]: Can we use the similar assertion to `expect(screen.getAllByRole('link')).toHaveLength(2);` here?

Comment (Quiddlee) [reply src/widgets/school-menu/ui/school-menu/school-menu.test.tsx]: fixed

File: src/widgets/school-menu/ui/school-menu/school-menu.test.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      </SchoolMenu>,
+    );
+
+    expect(screen.getAllByRole('link')).toHaveLength(2);
+
+    const links = screen.getAllByRole('link');
+
+    links.forEach((link) => {
+      expect(link).toBeInTheDocument();
+    });
```
[the remark this answers is not in the document]
Comment (Quiddlee) [reply src/widgets/school-menu/ui/school-menu/school-menu.test.tsx]: fixed

File: src/widgets/school-menu/ui/school-item/school-item.module.scss [hunk 3]
```diff
@@ -0,0 +1,70 @@
+.school-item {
+  display: flex;
+  gap: 5px;
+  column-gap: 15px;
+  align-items: center;
+
+  opacity: 1 !important;
```
[the remark this answers is not in the document]
Comment (Quiddlee) [reply src/widgets/school-menu/ui/school-item/school-item.module.scss]: fixed

**Comment by Quiddlee** [comment]
run visual now

**Comment by Quiddlee** [comment]
run visual now

**Comment by Quiddlee** [comment]
run visual now

**Comment by Quiddlee** [comment]
> Would be nice to take screenshots for the desktop menu as well https://percy.io/8d310036/web/rs-school-standalone/builds/37627699/changed/2047524732?browser=chrome&browser_ids=59%2C60%2C62%2C63&category=unchanged&device_name=&group_snapshots_by=similar_diff&os=&other-browser=&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=375&widths=375%2C1280

Done

https://percy.io/8d310036/web/rs-school-standalone/builds/37644869/changed/2048380025?browser=chrome&browser_ids=59%2C60%2C62%2C63&group_snapshots_by=similar_diff&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280

**Review by SpaNb4 — APPROVED** [review]

**Review by KristiBo — APPROVED** [review]

File: src/widgets/school-menu/ui/school-item/school-item.tsx [hunk 4]
```diff
@@ -1,63 +1,50 @@
...
+  ...props
+}: SchoolItemProps) => {
   return (
-    <li key={'id' in item ? item.id : item.title}>
-      <Link
-        href={item.detailsUrl}
-        className={'id' in item ? 'school-item with-icon' : 'school-item'}
-      >
-        {'iconSmall' in item && (
+    <li key={title} {...props}>
```
[the remark this answers is not in the document]
Comment (Quiddlee) [reply src/widgets/school-menu/ui/school-item/school-item.tsx:28]: removed

File: src/widgets/school-menu/ui/school-item/school-item.module.scss [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      flex-direction: column;
+      gap: 5px;
+      align-items: flex-start;
+      justify-content: flex-start;
+    }
+  }
+
+  .icon-wrapper {
+    grid-row: span 2;
+  }
```
[the remark this answers is not in the document]
Comment (Quiddlee) [reply src/widgets/school-menu/ui/school-item/school-item.module.scss:51]: removed

File: src/core/base-layout/components/footer/desktop-view.tsx [hunk 6]
```diff
@@ -9,11 +10,32 @@ export const DesktopView = async () => {
...
+        <SchoolMenu heading="rs school" color="light">
+          {schoolMenuStaticLinks.map((link, i) => (
+            <SchoolMenu.Item
+              key={i}
+              title={link.title}
+              description={link.description}
+              url={link.detailsUrl}
+              color="light"
+            />
+          ))}
```
[the remark this answers is not in the document]
Comment (Quiddlee) [reply src/core/base-layout/components/footer/desktop-view.tsx:22]: the menu is static, the order will never change

File: src/widgets/school-menu/ui/school-item/school-item.tsx [hunk 7]
```diff
@@ -1,63 +1,48 @@
...
-        {descriptionBlock}
+    <li {...props}>
+      <Link href={url} className={cx('school-item')}>
+        {icon && <Image
+          src={icon}
+          alt=""
+          width={32}
+          height={32}
+          data-testid="school-item-icon"
+        />}
```
[the remark this answers is not in the document]
Comment (Quiddlee) [reply src/widgets/school-menu/ui/school-item/school-item.tsx]: aria hidden attribute has been added

**Comment by Quiddlee** [comment]
run visual now

**Pull request merged**